### ISSUE TITLE
The Client repository is gone from the docker hub

### ISSUE THREAD

**Issue by adamradocz** [BOINC/boinc-client-docker#6]
The client repository is gone from the docker hub. Was it intentional?

**Comment by marius311** [comment]
Yea, I deleted it in preparation to move the build to the Docker Cloud, which is needed since we need to use build hooks, but I'm having trouble doing so (see https://forums.docker.com/t/cant-create-autobuild-for-github-organization-i-am-a-member-of/46852)

**Comment by adamradocz** [comment]
No, I'm not.

**Comment by adamradocz** [comment]
Are you sure, is it necessary to use the Docker Cloud? In assume the build older in the Docker Hub is the same as the build tags. So the first build will be the master.

**Comment by marius311** [comment]
Unfortunately I don't think so, in my testing it looks like the build order on both Hub and Cloud is random each time, at least when you're building from different Dockerfiles on the same branch (maybe what you had before with different branches it always built master first?) I think we'll get it figured out, my guess is one of the BOINC organization admins just needs to approve the third party app like I asked in the other thread.

**Comment by adamradocz** [comment]
Then add @davidpanderson to the team at Docker hub, maybe he can create the automated repo.

**Comment by marius311** [comment]
Looks like that may the only way. @davidpanderson 

1) Tell me your Docker Hub username so I can add you to the organization. 

2) Log in, then go to https://cloud.docker.com/app/boinc/settings and under Source Providers -> Github click the Connect Provider button and link your Github account. 

3) Then go to https://cloud.docker.com/app/boinc/repository/create, put `client` for the name, give it any description, then under Build Settings click the Github icon which should now be linked and should give you something that looks like this:

![image](https://user-images.githubusercontent.com/1510968/36706194-1812ed7e-1b69-11e8-8594-b917ff0080dc.png)

Hopefully from your account you can select "BOINC" as the organization and the boinc-client-docker repo as the repository.

**Comment by davidpanderson** [comment]
My name on Docker is DavidAnderson.
Let me know when you've added me.
-- D

**Comment by adamradocz** [comment]
I added you to the team.

**Comment by adamradocz** [comment]
Did you get the invitation?

**Comment by marius311** [comment]
@davidpanderson can you please take a look? We can't do much more until you link the repo.

**Comment by marius311** [comment]
If we don't get a response from David in the next day or so, given the difficulties getting this set up, what do you think about moving this out of the BOINC organization and back into your repo and working from there?

**Comment by TheAspens** [comment]
Marius - my id is theaspens.  If you add me I can follow the instructions above.  Thanks.

**Comment by marius311** [comment]
Thanks, Kevin. Just added you.

**Comment by TheAspens** [comment]
So if I link my account, it wants access to my personal repos (including private ones) as well as access to the BOINC organization and the IBM organization.  I cannot allow that level access.

Is there a way to create service type account on github and on docker that can provide that linkage so that it is not done using personal accounts?

If need be, I can create a different account on github and give it access, but I was wondering if there was guidance on the best way to set that up.

**Comment by marius311** [comment]
Thanks for trying, man that is annoying. As much as I love Docker, "Cloud" does not seem in the best shape, aside from that it has the build order feature that we need... (btw here some others raising some similar concerns I think https://forums.docker.com/t/question-about-docker-cloud-and-github-account-linking/27914/5)

I can't say I have any better suggestions than what you just suggested.

**Comment by marius311** [comment]
Or, so as to avoid mucking too much with Github accounts and the BOINC organization, we just nix automated builds until Cloud gets their stuff together and we can keep the repo on Cloud but just manually build / push after each commit? I don't anticipate this repo is going to get *that* much commit traffic that it'll be too prohibitive. Thoughts @adamradocz  ?

**Comment by adamradocz** [comment]
I think, we should move back to docker hub, in the worst case scenario we manually trigger the build, when significant changes made.

**Comment by marius311** [comment]
I'd prefer being able to manually build and push only because you never know how long the builds are going to stay in the queue (esp on Hub which is slower). So if you commit something you have to wait to see which build starts first, then if the base image didn't start first you have to wait for it to build, *then* trigger the dependent images. Since *something* will have to be done by hand, I prefer to just be able to build the images locally and push them, I can even make a script for that and commit it. 

So in conclusion, lets go with Hub but no automated build at all for now?

**Comment by davidpanderson** [comment]
Sorry, I've been sick recently.
I'll take a look at this today.
-- D

**Comment by davidpanderson** [comment]
I followed Marius's directions and everything seemed to work.
-- D

**Comment by marius311** [comment]
Looks to be working now, thanks @davidpanderson.
